examples/expositional/frameworks/cortexchat/cortex_chat_quickstart.ipynb

```diff
@@ -0,0 +1,303 @@
+    "                if decoded_line.startswith('data:'):\n",
+    "                    try:\n",
+    "                        data = json.loads(decoded_line[5:])\n",
+    "                        if data['delta']['content'][0]['type'] == \"text\":\n",
+    "                            print(data['delta']['content'][0]['text']['value'], end = '')\n",
+    "                            text += data['delta']['content'][0]['text']['value']\n",
+    "                        if data['delta']['content'][0]['type'] == \"citation\":\n",
+    "                            citation = data['delta']['content'][0]['citation']\n",
+    "                        if data['delta']['content'][0]['type'] == \"debug_info\":\n",
+    "                            debug_info = data['delta']['content'][0]['debug_info']\n",
+    "                            print(debug_info)\n",
+    "                    except json.JSONDecodeError:\n",
+    "                        print(f\"Error decoding JSON: {decoded_line}\")\n",
+    "                        print(f\"Previous line: {previous_line}\")\n",
+    "                previous_line = decoded_line\n",
+    "\n",
+    "    @instrument           \n",
+    "    def chat(self, query: str) -> tuple[str, str]:\n",
+    "        \"\"\"\n",
+    "        Sends a chat query to the Cortex Chat API and returns the response.\n",
+    "        Args:\n",
+    "            query (str): The chat query to send.\n",
+    "        Returns:\n",
+    "            tuple: A tuple containing the text response and citation.\n",
+    "        Raises:\n",
+    "            None\n",
+    "        Example:\n",
+    "            >>> cortex = CortexChat()\n",
+    "            >>> response = cortex.chat(\"Hello, how are you?\")\n",
+    "            >>> print(response)\n",
+    "            (\"I'm good, thank you!\", \"Cortex Chat API v1.0\")\n",
+    "        \"\"\"\n",
+    "\n",
+    "        url = self.url\n",
+    "        headers = {\n",
+    "            'X-Snowflake-Authorization-Token-Type': 'KEYPAIR_JWT',\n",
+    "            'Content-Type': 'application/json',\n",
+    "            'Accept': 'application/json',\n",
+    "            'Authorization': f\"Bearer {os.environ.get('SNOWFLAKE_JWT')}\"\n",
+    "        }\n",
+    "        data = {\n",
+    "            \"query\": query,\n",
+    "            \"model\": self.model,\n",
+    "            \"debug\": True,\n",
+    "            \"search_services\": [{\n",
+    "                \"name\": \"JOSH_DB.DATA.JOSH_CORTEX_SEARCH_SERVICE\",\n",
+    "                \"max_results\": 10,\n",
+    "            }],\n",
+    "            \"prompt\": \"{{.Question}} {{.Context}}\",\n",
+    "        }\n",
+    "\n",
+    "        response = requests.post(url, headers=headers, json=data, stream=True)\n",
+    "        if response.status_code == 200:\n",
+    "            text, citation, _ = self._handle_cortex_chat_response(response)\n",
+    "            return text, citation\n",
+    "        else:\n",
```

Comment:
same here

Reply:
We don't need citation for trulens info (retrieved context is in debug info), so not adding an exception if not provided by chat api.